### PR TITLE
Update category-header.tpl

### PR DESCRIPTION
**Pull request by rodriciru**
close https://github.com/PrestaShop/hummingbird/issues/476

### CODE REVIEW

**Review by sudo-do — DISMISSED** [review]
Since I don't see a reason to get invalid index in line 14.

File: templates/catalog/_partials/category-header.tpl [hunk 1]
```diff
@@ -9,7 +9,7 @@
             {if $category.description}
               <div id="category-description" class="rich-text mb-4">{$category.description nofilter}</div>
             {/if}
-            {if !empty($category.image.large.url)}
+            {if !empty($category.image}
```
Comment (mflasquin) [inline templates/catalog/_partials/category-header.tpl:12]:
```suggestion
            {if !empty($category.image) && !empty($category.image.large.url)}
```

**Review by mflasquin — CHANGES_REQUESTED** [review]

File: templates/catalog/_partials/category-header.tpl (hunk 1 above)
Comment (sudo-do) [reply templates/catalog/_partials/category-header.tpl:12]: isn't large url exist if image exists?

**Review by rodriciru — COMMENTED** [review]
Not necessary IMO

**Comment by rodriciru** [comment]
I dont know how to reject this changes, so i hope i do it ok

**Comment by sudo-do** [comment]
yeah, the change is not nessesary as far as i know prestashop environment

**Review by Hlavtox — APPROVED** [review]

**Comment by Hlavtox** [comment]
@rodriciru Can you show what error you are getting? I deleted the images from the category but I get no error 🤔

**Comment by kpodemski** [comment]
@rodriciru that's odd, `empty` for this key should be safe 🤔  have you checked this file 078ccc9839dcacadc15c0f4aa16f402af5079aa_2.file.category.tpl.php on line 57 to see if it is really about this line?

**Comment by rodriciru** [comment]
> @rodriciru that's odd, `empty` for this key should be safe 🤔 have you checked this file 078ccc9839dcacadc15c0f4aa16f402af5079aa_2.file.category.tpl.php on line 57 to see if it is really about this line?

Yep, it's weird as hell, but I have some answers.
Looking at compiled PHP, I see that some empty functions translate to simple `if`'s and others don't.
Concretely, this image `empty `translates to `if `and the `empty `2 lines above translate to an `empty `instead of an `if`:
**PLEASE KEEP IN MIND It's NOT EXACTLY THIS FILE, BUT IT'S ALMOST THE SAME**
```
            <?php if ($_smarty_tpl->tpl_vars['category']->value['image']) { ?>
                <div class="category-cover">
                  <img src="<?php echo htmlspecialchars((string) $_smarty_tpl->tpl_vars['category']->value['image']['large']['url'], ENT_QUOTES, 'UTF-8'); ?>
" alt="<?php if (!empty($_smarty_tpl->tpl_vars['category']->value['image']['legend'])) {
                  echo htmlspecialchars((string) $_smarty_tpl->tpl_vars['category']->value['image']['legend'], ENT_QUOTES, 'UTF-8');
                } else {
                  echo htmlspecialchars((string) $_smarty_tpl->tpl_vars['category']->value['name'], ENT_QUOTES, 'UTF-8');
                } ?>">
                </div>
            <?php } ?>
```
Effectively, `empty` not raises any warnings, but `if` does, so this is why we get the warning.
Still, I think this is an easy and no harmful at all fix, even if the blame of the issue is smarty

**Comment by leemyongpakvn** [comment]
@florine2623 FYI, hummingbird is different from classic-theme. There is no `_dev` directory in hummingbird, you need to create a .env file inside its webpack directory, and run `npm run build` from hummingbird directory following this _cookbook_ https://github.com/PrestaShop/hummingbird#how-to-build-assets ;)

**Comment by florine2623** [comment]
Hello @rodriciru ,

It seems like no one can reproduce your issue, neither @Hlavtox , @kpodemski nor @leemyongpakvn .

Could you explain further so it can be reproduced and tested ?

**Comment by leemyongpakvn** [comment]
As usual, QA will close the issue if they can not reproduce it :ok_hand:

**Comment by Hlavtox** [comment]
@rodriciru I will try to reproduce once again, if you really say it happens. There is no reason for smarty to convert empty to if. If it does so, it's stupid. 🤔

On what core version you are getting this behavior?

**Comment by Prestaworks** [comment]
{if !empty($category.image.large.url)} is also present in classic theme, I tried deleting the category images, "image" => null shows on the category dump, tested on PS 8.1.2 and php 7.2, 7.4 and 8.1 still nothing in the php logs or Prestashop log files.

so if this is caused by "image" not existing it's maybe an older PS version?
To make the check safe it should maybe be something like
{if !empty($category.image) && !empty($category.image.large)} 

But I can't recreate the error on PS 8.1.2.

**Comment by rodriciru** [comment]
I don't remember exactly the version, but I try to keep the latest version
of ps on my stores. So by the time the issue is created, the most recent
version at that time.

Anyway,seeing I'm the only,and can't reproduce by any of yours, let's close
and will open a new one if happens again

**Comment by kpodemski** [comment]
@Hlavtox @Prestaworks maybe it was a bug in some Smarty version because I noticed that problem as well, I've had `{if !empty($array.key)}` it started flooding logs with a warnings/notice.

**Comment by kpodemski** [comment]
@rodriciru 1.7.6 something

**Comment by florine2623** [comment]
I suggest closing this issue as it cannot be reproduced https://github.com/PrestaShop/hummingbird/issues/476#issuecomment-1864780014

**Pull request closed without merging**